src/map.h

```diff
@@ -161,6 +162,60 @@ struct bash_params {
     bool bashed_solid; // Did we bash furniture, terrain or vehicle
 };
 
+// For light we store four values, depending on the direction that the light
+// comes from.  This allows us to determine whether the side of the wall the
+// player is looking at is lit.
+// For non-opaque tiles direction doesn't matter so we just use the single
+// default_ value.
+enum class quadrant {
+    NE, SE, SW, NW,
+    default_ = NE
+};
+
+struct four_quadrants {
+    four_quadrants() = default;
+    explicit constexpr four_quadrants( float v ) : values{{v, v, v, v}} {}
+
+    std::array<float, 4> values;
+
+    float &operator[]( quadrant q ) {
+        return values[static_cast<int>( q )];
+    }
+    float operator[]( quadrant q ) const {
+        return values[static_cast<int>( q )];
+    }
+    void fill( float v ) {
+        std::fill( values.begin(), values.end(), v );
+    }
+    float max() const {
+        return *std::max_element( values.begin(), values.end() );
+    }
+
+    friend inline four_quadrants operator*( const four_quadrants &l, const four_quadrants &r ) {
```

Comment:
`inline` is redundant here. Functions defined inside the class definition are automatically `inline`.

Reply:
I thought that was only true of member functions, but it seems you're right; it applies to friend definitions also.  Thanks.